### ISSUE TITLE
TODO list

### ISSUE THREAD

**Issue by armleo** [armleo/OpenLane#24]
🚫 = Not done yet
✔️ = Done
🗑️ = No longer needed
⏸️ = Pause/Pending

# Arman's tasks

✔️ Routing obstruction for DRCs. Tried, did not work. `This layer can't abut or partially overlap between subcells` error on every standard cell. UPD: Tim Edwards said that this is not an issue as this is an issue in OpenPDKs and DRC check with LEF abstracts of standard cells.
🗑️ Create an issue regarding above
✔️ Reference the newly created issue in the documentation
✔️ #23 1. Simplify intro
🗑️ #23 2. What is relevant section?
🗑️ #23 3. json vs tcl?
✔️ #23 4. No promises
✔️ #23 5.1. name collisions. P.S.: Why is there two number 5 actions? 
⏸️ #23 6.1 absolute path is used [Pending on the issue]
✔️ #23 6.2 what is the meaning of this what is subcomponent. Discussed in the meeting
✔️ #23 6.3 Please use suitable demarcations
⏸️ #23 6.4 "issue the warning to use the right pathnames". No, I want to mention the error. How else am I going, to refer to it?
✔️ #23 7.1 There is an abrupt reference to Caravel. Pending to discuss.
✔️ #23 7.2 USE_POWER_PINS relevant warning needs to be removed
✔️ #23 8.1 separate line for readability
✔️ #23 8.2 refrain from storytelling. Need to discuss.
✔️ #23 8.3 Rephrase escapes and stuff
🗑️ #23 8.4 Make warning about LVS issue a warning
✔️ #23 8.5/8.6 confusion regarding the slash and lvs issues
✔️ #23 9.1 Convert information about DIE AREA to note. Done
✔️ #23 9.2 Regarding FP_PDN_HORIZONTAL_HALO and FP_PDN_VERTICAL_HALO
✔️ #23 10. Rewrite for clarity the macro placement
✔️ #23 Second 10.  Change title to Common Problems to Avoid
✔️ #23 11. Before hardening the macro block define the I/O pin placement. Moved to will not be done, because no time to rewrite significant part of the text. Added a reference to existing guide.
✔️ #23 12. [create-a-new-design](https://armleo-openlane.readthedocs.io/en/merge-window-4/docs/source/openram.html#create-a-new-design) remove . at the end of sub-heading.
✔️ #23 Karims' feedback on QUIT_ON_MAGIC_DRC 
✔️ #23 Karims' feedback on VERILOG_FILES_BLACKBOX
✔️ #18 Go over one by one. Adding the issue to this list and fixing it
✔️ #18 Installation steps that are specific to Ubuntu/Linux need to be moved to a common location and removed from others
✔️ #18 Instructions for enabling docker at boot
✔️ #18 Document running the terminal
✔️ Remove verilog blackboxing from OpenRAM
✔️ Add EXTRA_LIBS section
✔️ Remove SYNTH_USE_PG_PINS_DEFINES/USE_POWER_PINS
✔️ Create an issue regarding absolute paths in config.json
✔️ Separate section for tutorials
✔️ Document different integration ways of BB
🗑️ Document different integration ways of PDN macro hooks
✔️ Add reference to IO placement in hardening macros
✔️ Define what IS_CORE and is not core is. match the documentation
✔️ Add ring around macro comparison picture
✔️ Add overview of running the flow on the design in hierarchical guide
✔️ Added missing RTL file creation process in the hierarchical guide.
✔️ Fix all TODOs inside OpenRAM tutorial
✔️ Fix all TODOs inside hier design flow tutorial
🗑️ Add see also at the end of every new page. No longer needed. There is a "Next" button instead.
✔️ Add macOS overview
✔️ Resolve #14 regarding line 8
✔️ Resolve #14 regarding line 10
✔️ Resolve #14 regarding line 20
✔️ Resolve #14 regarding line 194
✔️ Resolve #14 regarding line 268/275
✔️ Resolve #14 regarding line 291
✔️ Resolve #14 regarding line 346
✔️ point to the use of odb
✔️ Resolve merge conflicts
✔️ No promises for the future direction
✔️ Make sure every page has an overview
✔️ Make sure there is no TODOs
✔️ Use active voice
✔️ Make sure there is no missing links
✔️ Spellcheck Ubuntu install
✔️ Spellcheck macOS install
✔️ Spellcheck Windows install
✔️ Spellcheck Linux install
✔️ Spellcheck updating
✔️ Spellcheck hierarchical design flow
✔️ Spellcheck OpenRAM
✔️ Create an issue regarding: "Document which vars are arrays and which are not"

# NotArman's Tasks

Final touches for mentors
🚫 Make sure there is no TODOs
🚫 Use active voice
🚫 No promises for the future direction
🚫 Make sure every page has an overview
🚫 Review #6

Things not part of this project:
🚫 PDN straps configuration
🚫 Fix link from readme to openlane docs
🚫 Straps for smaller macros
🚫 Debugging power connections
🚫 Flow diagram update
🚫 Integrating fully custom analog cell
🚫 Document which vars are arrays and which are not
🚫 GitHub CI for Docs
🚫 Before hardening the macro block define the I/O pin placement.
🚫 Dcoumenting generic PNR flow
🚫 Document issues with links in rst
🚫 Fix issues with links in rst
